treeherder/webapp/api/infra_compare.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import time
+
+from rest_framework import generics
+from treeherder.model import models
+
+from .infra_serializers import InfraCompareSerializer, InfraCompareQuerySerializers
+
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+
+class InfraCompareView(generics.ListAPIView):
+    serializer_class = InfraCompareSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = InfraCompareQuerySerializers(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+
+        startday = query_params.validated_data['startday']
+        endday = query_params.validated_data['endday']
+        project = query_params.validated_data['project']
+        revision = query_params.validated_data['revision']
+        interval = query_params.validated_data['interval']
```

Comment:
can we validate some of the params here and ensure that interval is something reasonable 1<x<100?

maybe same for others, project in (...), startday/endday in range(...)

Reply:
I think interval of 1<x<=86400 will do since atleast we will be comparing pushes from a day

Reply:
sold